### DISCUSSION TITLE
Map keys to delete word and line next to cursor

### DISCUSSION THREAD

**Discussion by cg00001** [kovidgoyal/kitty#5431] [Q&A] [answered]
I want to use ctrl + delete to delete line next to cursor, and alt + delete to delete word after cursor.
So far I found how to delete word before cursor:
map alt+backspace send_text all \x17

Can someone help?

**Comment by cg00001** [comment]
So far I only found how to delete the whole line. I tried from \x01 to \x45 -I know past 31 has no meaning.
```
# Delete whole line
map ctrl+delete send_text all \x15
```
Seems documentation on control codes is foggy on all sites I searched. The equivalent to Ctrl + K, doesn't seem to exist.

**Comment by kovidgoyal** [comment, marked as answer]
This doesnt have anything to do with control codes. You need to figure out what key the shell you are using has mapped to the delete line or delete word function, and either map another key in the shell's config or map the key in kittys config to send the bytes corresponding to the key the shell expects, which you can get using showkey or kitty +kitten show_key

**Reply by cg00001** [reply]
> This doesnt have anything to do with control codes. You need to figure out what key the shell you are using has mapped to the delete line or delete word function, and either map another key in the shell's config or map the key in kittys config to send the bytes corresponding to the key the shell expects, which you can get using showkey or kitty +kitten show_key

Thank you.
Using kitty +kitten show_key with Ctrl + K and Alt + D gave the codes.

```
# Delete word after cursor
map alt+delete send_text all \x1bd

# Delete all words after cursor
map ctrl+delete send_text all \x0b
```
